test/modules/DirectMethodReceiver/src/DirectMethodReceiver.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace DirectMethodReceiver
+{
+    using System;
+    using System.Net;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    class DirectMethodReceiver : IDisposable
+    {
+        Guid batchId;
+        IConfiguration configuration;
+        long directMethodCount = 1;
+        ILogger logger;
+        ModuleClient moduleClient;
+        Option<TestResultReportingClient> testResultReportingClient;
+        Option<string> trackingId;
+
+        public DirectMethodReceiver(
+            ILogger logger,
+            IConfiguration configuration)
+        {
+            Preconditions.CheckNotNull(logger, nameof(logger));
+            Preconditions.CheckNotNull(configuration, nameof(configuration));
+
+            Option<Uri> testReportCoordinatorUrl = Option.Maybe(configuration.GetValue<Uri>("testResultCoordinatorUrl"));
+            testReportCoordinatorUrl.ForEach(
+                (Uri uri) => this.testResultReportingClient = Option.Some<TestResultReportingClient>(new TestResultReportingClient { BaseUrl = uri.AbsoluteUri }),
+                () => this.testResultReportingClient = Option.None<TestResultReportingClient>());
+
+            this.logger = logger;
+            this.configuration = configuration;
+            this.batchId = Guid.NewGuid();
+            this.trackingId = Option.Maybe(this.configuration.GetValue<string>("trackingId"));
```

Comment:
You can validate if testReportCoordinatorUrl have value; therefore below usage, you can call OrDefault() instead of Expect.